onnxruntime/core/providers/openvino/openvino_provider_factory.cc

```diff
@@ -63,12 +56,81 @@ struct OpenVINO_Provider : Provider {
   void* GetInfo() override { return &g_info; }
 
   std::shared_ptr<IExecutionProviderFactory> CreateExecutionProviderFactory(const void* void_params) override {
-    auto& params = *reinterpret_cast<const OrtOpenVINOProviderOptions*>(void_params);
-    return std::make_shared<OpenVINOProviderFactory>(params.device_type, params.enable_vpu_fast_compile,
-                                                     params.device_id, params.num_of_threads,
-                                                     params.cache_dir,
-                                                     params.context, params.enable_opencl_throttling,
-                                                     params.enable_dynamic_shapes);
+    auto& provider_options_map = *reinterpret_cast<const ProviderOptions*>(void_params);
+
+    const char* device_type = "";           // [device_type]: Overrides the accelerator hardware type and precision
+                                            //   with these values at runtime.
+    bool enable_vpu_fast_compile = false;   // [enable_vpu_fast_compile]: Fast-compile may be optionally enabled to
+                                            // speeds up the model's compilation to VPU device specific format.
+    const char* device_id = "";             // [device_id]: Selects a particular hardware device for inference.
+    size_t num_of_threads = 8;              // [num_of_threads]: Overrides the accelerator default value of number of
+                                            //  threads with this value at runtime.
+    const char* cache_dir = "";             // [cache_dir]: specify the path to
+                                            // dump and load the blobs for the model caching/kernel caching (GPU)
+                                            // feature. If blob files are already present, it will be directly loaded.
+    int num_streams = 1;                    // [num_streams]: Option that specifies the number of parallel inference
```

Comment:
Will adding num_stream
 lead to backward compatibility issue

Reply:
A mapping is added for legacy API. 
By default its set to 1. 
> ov_options_converted_map["num_streams"] = "1"